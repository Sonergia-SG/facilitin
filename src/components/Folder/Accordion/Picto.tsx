import React from 'react';

import { CheckPoint } from '../../../store/reducer/entities/types';

interface Props {
  checkPoint: CheckPoint;
}

const resolveIco = (c: CheckPoint) => {
  if (c.id_penalite === 2 && c.pivot.valide === 0) {
    return {
      key: 'error',
      name: 'fa-exclamation-triangle',
      color: '#F61616',
    };
  }

  if (c.id_penalite === 1 && c.pivot.valide === 0) {
    return {
      key: 'warning',
      name: 'fa-exclamation-triangle',
      color: '#FBD44A',
    };
  }

  if (c.pivot.valide === -1) {
    return {
      key: 'untraited',
      name: 'fa-circle',
      color: '#444',
    };
  }

  if (c.pivot.valide === 1) {
    return {
      key: 'ok',
      name: 'fa-check-circle',
      color: '#2BA048',
    };
  }

  return {
    key: 'untraited',
    name: 'fa-circle',
    color: '#444',
  };
};

const Picto = ({ checkPoint }: Props) => {
  const icoConfig = resolveIco(checkPoint);

  return (
    <div key={icoConfig.key}>
      <i className={`fas ${icoConfig.name}`} style={{ color: icoConfig.color }} />
    </div>
  );
};

export default Picto;
