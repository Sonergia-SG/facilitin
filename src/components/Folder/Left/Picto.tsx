import React from 'react';

interface Props {
  total: number;
  valid: number;
  litige?: boolean;
}

const resolveIco = (total: number, valid: number, litige?: boolean) => {
  if (litige) {
    return {
      key: 'error',
      name: 'fa-exclamation-triangle',
      color: '#F61616',
    };
  }

  if (valid === total) {
    return {
      key: 'ok',
      name: 'fa-check-circle',
      color: '#2BA048',
    };
  }

  return {
    key: 'warning',
    name: 'fa-exclamation-triangle',
    color: '#FBD44A',
  };
};

const Picto = ({ litige, total, valid }: Props) => {
  const icoConfig = resolveIco(total, valid, litige);

  return (
    <div key={icoConfig.key}>
      <i className={`fas ${icoConfig.name}`} style={{ color: icoConfig.color }} />
    </div>
  );
};

export default Picto;
