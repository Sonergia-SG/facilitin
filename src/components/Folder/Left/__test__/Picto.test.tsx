import React from 'react';
import { shallow } from 'enzyme';

import Picto from '../Picto';

describe('Picto', () => {
  it('render warning for litige', () => {
    const wrapper = shallow(<Picto total={2} valid={1} litige={1} rejected={0} unTraited={0} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('render error for rejected', () => {
    const wrapper = shallow(<Picto total={2} valid={1} litige={0} rejected={1} unTraited={0} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('render untraited', () => {
    const wrapper = shallow(<Picto total={2} valid={1} litige={0} rejected={0} unTraited={1} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('render success if total === valid', () => {
    const wrapper = shallow(<Picto total={2} valid={2} litige={0} rejected={0} unTraited={0} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('make error more priority', () => {
    const wrapper = shallow(<Picto total={4} valid={1} litige={1} rejected={1} unTraited={1} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('make warning more priority', () => {
    const wrapper = shallow(<Picto total={4} valid={2} litige={1} rejected={0} unTraited={1} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('make untraited more priority', () => {
    const wrapper = shallow(<Picto total={4} valid={3} litige={0} rejected={0} unTraited={1} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('make checked less priority', () => {
    const wrapper = shallow(<Picto total={4} valid={4} litige={0} rejected={0} unTraited={0} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('display untraited for strange results', () => {
    const wrapper = shallow(<Picto total={4} valid={5} litige={0} rejected={0} unTraited={0} />);

    expect(wrapper).toMatchSnapshot();
  });
});
