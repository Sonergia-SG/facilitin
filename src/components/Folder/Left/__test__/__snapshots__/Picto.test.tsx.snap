// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picto display untraited for strange results 1`] = `
<div
  key="untraited"
>
  <i
    className="fas fa-circle"
    style={
      Object {
        "color": "#444",
      }
    }
  />
</div>
`;

exports[`Picto make checked less priority 1`] = `
<div
  key="ok"
>
  <i
    className="fas fa-check-circle"
    style={
      Object {
        "color": "#2BA048",
      }
    }
  />
</div>
`;

exports[`Picto make error more priority 1`] = `
<div
  key="error"
>
  <i
    className="fas fa-exclamation-triangle"
    style={
      Object {
        "color": "#F61616",
      }
    }
  />
</div>
`;

exports[`Picto make untraited more priority 1`] = `
<div
  key="untraited"
>
  <i
    className="fas fa-circle"
    style={
      Object {
        "color": "#444",
      }
    }
  />
</div>
`;

exports[`Picto make warning more priority 1`] = `
<div
  key="warning"
>
  <i
    className="fas fa-exclamation-triangle"
    style={
      Object {
        "color": "#FBD44A",
      }
    }
  />
</div>
`;

exports[`Picto render error for rejected 1`] = `
<div
  key="error"
>
  <i
    className="fas fa-exclamation-triangle"
    style={
      Object {
        "color": "#F61616",
      }
    }
  />
</div>
`;

exports[`Picto render success if total === valid 1`] = `
<div
  key="ok"
>
  <i
    className="fas fa-check-circle"
    style={
      Object {
        "color": "#2BA048",
      }
    }
  />
</div>
`;

exports[`Picto render untraited 1`] = `
<div
  key="untraited"
>
  <i
    className="fas fa-circle"
    style={
      Object {
        "color": "#444",
      }
    }
  />
</div>
`;

exports[`Picto render warning for litige 1`] = `
<div
  key="warning"
>
  <i
    className="fas fa-exclamation-triangle"
    style={
      Object {
        "color": "#FBD44A",
      }
    }
  />
</div>
`;
